packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/mappers.ts

```diff
@@ -47,43 +47,51 @@ interface ToCoreStakePoolInput {
-    const toReturn: Cardano.StakePool = {
-      cost: poolData.cost,
-      epochRewards: poolRewards.filter((r) => r.hashId === poolData.hashId).map((reward) => reward.epochReward),
-      hexId: poolData.hexId,
-      id: poolData.id,
-      margin: poolData.margin,
-      metrics:
-        poolMetrics.find((metrics) => metrics.hashId === poolData.hashId)?.metrics || ({} as Cardano.StakePoolMetrics),
-      owners: poolOwners.filter((o) => o.hashId === poolData.hashId).map((o) => o.address),
-      pledge: poolData.pledge,
-      relays: poolRelays.filter((r) => r.updateId === poolData.updateId).map((r) => r.relay),
-      rewardAccount: poolData.rewardAccount,
-      status: getPoolStatus(registrations[0], lastEpoch, retirements[0]),
-      transactions: {
-        registration: registrations.map((r) => r.transactionId),
-        retirement: retirements.map((r) => r.transactionId)
-      },
-      vrf: poolData.vrfKeyHash
-    };
-    if (poolData.metadata) toReturn.metadata = poolData.metadata;
-    if (poolData.metadataJson) toReturn.metadataJson = poolData.metadataJson;
-    return toReturn;
-  }),
+export const toCoreStakePool = (
+  poolHashIds: number[],
+  {
+    poolOwners,
+    poolDatas,
+    poolRegistrations,
+    poolRelays,
+    poolRetirements,
+    poolRewards,
+    lastEpoch,
+    poolMetrics,
+    totalCount
+  }: ToCoreStakePoolInput
+): StakePoolSearchResults => ({
+  pageResults: poolHashIds
+    .map((hashId) => {
+      const poolData = poolDatas.find((data) => data.hashId === hashId);
+      if (!poolData) return;
+      const registrations = poolRegistrations.filter((r) => r.hashId === poolData.hashId);
+      const retirements = poolRetirements.filter((r) => r.hashId === poolData.hashId);
+      const toReturn: Cardano.StakePool = {
+        cost: poolData.cost,
+        epochRewards: poolRewards.filter((r) => r.hashId === poolData.hashId).map((reward) => reward.epochReward),
+        hexId: poolData.hexId,
+        id: poolData.id,
+        margin: poolData.margin,
+        metrics:
+          poolMetrics.find((metrics) => metrics.hashId === poolData.hashId)?.metrics ||
+          ({} as Cardano.StakePoolMetrics),
+        owners: poolOwners.filter((o) => o.hashId === poolData.hashId).map((o) => o.address),
+        pledge: poolData.pledge,
+        relays: poolRelays.filter((r) => r.updateId === poolData.updateId).map((r) => r.relay),
+        rewardAccount: poolData.rewardAccount,
+        status: getPoolStatus(registrations[0], lastEpoch, retirements[0]),
+        transactions: {
+          registration: registrations.map((r) => r.transactionId),
+          retirement: retirements.map((r) => r.transactionId)
+        },
+        vrf: poolData.vrfKeyHash
+      };
+      if (poolData.metadata) toReturn.metadata = poolData.metadata;
+      if (poolData.metadataJson) toReturn.metadataJson = poolData.metadataJson;
+      return toReturn;
+    })
+    .filter((result) => result !== undefined) as Cardano.StakePool[],
```

Comment:
What is the case where `result` is _undefined_ here? At first look it seems we iterate over the collection `poolHashIds` (passed as `orderedResultHashIds`  which are previously filled with additional [restOfPoolData](https://github.com/input-output-hk/cardano-js-sdk/pull/270/files#diff-5ad8b26faabe50605214aec922ed96428a741e5e07da541676062cec7a693a17R74) ) which is bigger than `poolDatas`